### PR TITLE
Add rough outline of AddTaskCommand into Developer's guide

### PR DESCRIPTION
**Pull request by Kang-Quan**
Add rough outline of AddTaskCommand into Developer's guide

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -276,6 +276,65 @@ As a result, editing a supplier's details necessitate a corresponding update to
...
+
+**Steps**
+
+Step 1. The user enters the `addTask d/Buy Chicken dl/2020-12-12 t/Food` command.
+d/ represents the task description, dl/ represents deadline and t/ represents tag.
+
+Step 2. The `AddTaskCommandParser` parses the input and ensures that command is valid.
+For it to be valid, all compulsory field such as deadline and descriptions must be present.
+Tag field is optional. Further checks are that date given are in the correct format of 
+`yyyy-mm-dd`. If these checks are not valid, an exception will be thrown.
```
Comment (hauchongtang) [inline docs/DeveloperGuide.md:306]: First check is for whether is yyyy-mm-dd.
Second check is for whether the yyyy, mm or dd is valid as per Georgian calender, i.e. months must be 1 <= month <= 12 and day is 1-28/29/30/31 (Feb esp)

**Review by hauchongtang — COMMENTED** [review]
Missed out on another more specific chat !

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Kang-Quan) [reply docs/DeveloperGuide.md]: Noted, I have added the more specific check to the developer's guide

**Review by hauchongtang — APPROVED** [review]
LGTM

**Pull request merged**